Added Message_Spam_Detection script

Nkap23 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.
Help us understand your motivation by explaining why you decided to make this change.
Happy Contributing!
-->

# Description
Added script for Message_Spam_Detection
## Fixes #327 

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?
- [X] Yes
- [ ] No

## Type of change
- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] My code follows the style guidelines(Clean Code) of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have created a helpful and easy to understand `README.md`
- [X] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [X] My changes generate no new warnings
- [X] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

santushtisharma10 commented:
@Nkap23 you have to add python file not a notebook.

kaustubhgupta reviewed (changes requested):
As suggested by @santushtisharma10, @Nkap23 you have to add python script files and not the notebook files

Nkap23 commented:
> As suggested by @santushtisharma10, @Nkap23 you have to add python script files and not the notebook files

The notebook contains various steps, from data preprocessing to model training. 
Changing it to python script, would cause issues, cause all the steps have different requirements and thus are individual cells in the notebook.
When converted to scripts, they won't be able to function together properly.

If python script is really required, I can add a script along with notebook. But newcomers won't be able to understand the code.

kaustubhgupta commented:
@Nkap23 Take a look at this PR and you will get to know what we actually require

Nkap23 commented:
> @Nkap23 Take a look at this PR and you will get to know what we actually require:
> #556

okay. I will update the PR with python file.

kaustubhgupta commented:
@Nkap23 Just for the update, did you found the workaround?

Nkap23 commented:
> @Nkap23 Just for the update, did you found the workaround?

Hello. Yes, I have updated the PR. Request you to review it.
Thanks!

Nkap23 commented on Message_Spam_Detection/Message_Spam_Detection.py:
```diff
@@ -0,0 +1,143 @@
+y_train=y_train.astype(int)
+y_val=y_val.astype(int)
+
+#multinomial model
```
Okay! I have created a different script and added the saved model.

kaustubhgupta commented on Message_Spam_Detection/Dataset_Cleaning.py:
```diff
@@ -0,0 +1,77 @@
+
+#reading the dataset 
+#dataset: https://www.kaggle.com/uciml/sms-spam-collection-dataset
+msg=pd.read_csv("dataset.csv",encoding='latin-1')
```
There is no file named dataset.csv. Also, correct the path of the files would  be "./Message_Spam_Detection/filename"

Nkap23 replied:
Okay. Will make the changes!

kaustubhgupta commented on Message_Spam_Detection/Message_Spam_Detection.py:
```diff
@@ -0,0 +1,49 @@
+y=y.astype(int)
+
+#randomforstclassifier model
+model=RandomForestClassifier(n_estimators=100,random_state=0)
```
you need to do the model training in the cleaning file only and load the model in this file

Nkap23 replied:
The trained model has 5000 features (based on the training dataset).
The count vectorizer I used for model training and user input needs to be the same (then only the number of features would be the same)
If I use a different count vectorizer, the number of input features would be different and it will give an error. 
So that's why model training and user input are in same file.

kaustubhgupta replied:
okay got it, you can skip this suggestion and work on the previous ones

Nkap23 replied:
Okay! I have done the required changes!

kaustubhgupta reviewed (changes requested):
Umm, the model is not giving correct answers even for inputs for which it was trained (2nd one): 

![image](https://user-images.githubusercontent.com/43691873/111893172-baa99b80-8a26-11eb-8c45-22f0d5b4bf2f.png)

Nkap23 commented:
> Umm, the model is not giving correct answers even for inputs for which it was trained (2nd one):
> 
> ![image](https://user-images.githubusercontent.com/43691873/111893172-baa99b80-8a26-11eb-8c45-22f0d5b4bf2f.png)

okay! I have fixed it. I found the mistake. Need to clean the user input text also, before feeding it to the model. I have added that part in the code.

kaustubhgupta commented on Message_Spam_Detection/Message_Spam_Detection.py:
```diff
@@ -0,0 +1,80 @@
+model=RandomForestClassifier(n_estimators=100,random_state=0)
+model.fit(x,y)
+filename="randomforest.sav"
+pickle.dump(model,open(filename,"wb"))
```
I got the error: As you have trained the model here, the new model is not used in the prediction part. You are loading the old model uploaded in this PR. Now:
- delete the model uploaded in this PR
- don't save the model (pickel dump) and directly use the model while making prediction 
- Remove lines: 38, 39, 75 in this regard

Nkap23 replied:
Done!

antrikshmisri reviewed (approved):
LGTM 👍🏻

santushtisharma10 reviewed (approved):
@Nkap23 💯 💯

avinashkranjan reviewed (approved):
LGTM...Great Job @Nkap23
